### PR TITLE
pijul-devel: new port (1.0.0-beta.6)

### PR DESCRIPTION
**Pull request by TheRealKeto**
#### Description

Introduce `pijul-devel`, since the project is currently in beta; latest stable version available doesn't work with OpenSSL 3 and can't be built.

While I understand the rationale behind having the latest stable release, pijul has not had a stable release in at least 2 years. Some will say that this is not true, as evident by crates.io pointing to 0.12.2 being a stable release. This is true. It's also true that this version of the crate can no longer be built, as other versions of its dependencies are no longer buildable.

Thus, the version introduced by this commit should be accepted. While I cannot speak on the behalf of pijul's developers, this is the only distributable version that is not only available in some way (as the project itself is hard to get access to in archives, thanks to the fact that it is self-hosted), but also buildable.

Merging this would not only resolve the tickets attached to the commit, but would be a move towards resolving #20537, which specifically bumps all libsodium dependents.

Closes: https://trac.macports.org/ticket/65221
Closes: https://trac.macports.org/ticket/66884

###### Tested on

macOS 12.6 21G115 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by l2dy** [comment]
How about creating a `pijul-devel` port for the betas and remove this port for now?

**Comment by TheRealKeto** [comment]
This is possible; unsure how much more support is needed for this change, but considering the fact that this package doesn't have any dependents, it's a good idea.

File: devel/pijul-devel/Portfile [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+set pname           pijul
+name                ${pname}-devel
+version             1.0.0-beta.6
+categories          devel
+license             GPL-2+
+maintainers         nomaintainer
+description         ${pname} is a free and open source distributed version control system
+long_description    ${description}. Its distinctive feature is to be based on a sound theory, \
+                    which makes it easier to learn and use, and really distributed.
+homepage            https://${pname}.org/
```
Comment (l2dy) [inline devel/pijul-devel/Portfile:16]: Nitpick: There is no need to replace `pijul` with `${pname}` in homepage. It's a complete URL, not a common pattern where `${pname}` can be replaced with other names.

**Comment by l2dy** [comment]
Merged. Thanks!

**Pull request merged**